dashboard/modules/reporter/reporter_head.py

```diff
@@ -473,6 +473,136 @@ async def cpu_profile(self, req) -> aiohttp.web.Response:
         else:
             return aiohttp.web.HTTPInternalServerError(text=reply.output)
 
+    @routes.get("/memory_profile")
+    async def memory_profile(self, req) -> aiohttp.web.Response:
+        """
+        Retrieves the memory profile for a specific wrker or task.
+        Note that for task, one worker process works on one task at a time
+        or one worker works on multiple async tasks.
+
+        Args:
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the memory profile data.
+
+        Raises:
+            aiohttp.web.HTTPInternalServerError: If no stub
+            found from the given ip value
+            aiohttp.web.HTTPInternalServerError: If the
+            "task_id" parameter exists but either "attempt_number"
+            or "node id" is missing in the request query.
+            aiohttp.web.HTTPInternalServerError: If the maximum
+            duration allowed is exceeded.
+            aiohttp.web.HTTPInternalServerError If requesting task
+            profiling for the worker begins working on another task
+            during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is
+            an internal server error during the profile retrieval.
+        """
+        is_task = "task_id" in req.query
+
+        if is_task:
+            if "attempt_number" not in req.query:
+                return aiohttp.web.HTTPInternalServerError(
+                    text=(
+                        "Failed to execute task profiling: "
+                        "task's attempt number is required"
+                    )
+                )
+            if "node_id" not in req.query:
+                return aiohttp.web.HTTPInternalServerError(
+                    text=(
+                        "Failed to execute task profiling: "
+                        "task's node id is required"
+                    )
+                )
+
+            task_id = req.query.get("task_id")
+            attempt_number = req.query.get("attempt_number")
+            node_id = req.query.get("node_id")
+            ip = DataSource.node_id_to_ip[node_id]
+            try:
+                (pid, _) = await self.get_worker_details_for_running_task(
+                    task_id, attempt_number
+                )
+            except ValueError as e:
+                raise aiohttp.web.HTTPInternalServerError(text=str(e))
+        else:
+            pid = int(req.query["pid"])
```

Comment:
is it possible if `pid` is missing?

Reply:
it might be possible. In this case it will just throw internal server error 500 since profiling a profile will definitely require `pid`. Our frontend will always include `pid.